src/Microsoft.AspNet.Mvc.ViewFeatures/ViewComponents/ViewComponentMethodSelector.cs

```diff
@@ -55,42 +55,44 @@ public static MethodInfo FindSyncMethod(TypeInfo componentType, object[] args)
                 throw new InvalidOperationException(
                     Resources.FormatViewComponent_SyncMethod_ShouldReturnValue(SyncMethodName));
             }
+            else if (method.ReturnType.IsAssignableFrom(typeof(Task)))
+            {
+                throw new InvalidOperationException(
+                    Resources.FormatViewComponent_SyncMethod_CannotReturnTask(SyncMethodName, nameof(Task)));
```

Comment:
There's another usage for this on https://github.com/aspnet/Mvc/pull/3267#discussion-diff-41322297R34 where it prints `Task<T>` (without nameof etc). Just wanted these two to be consistent.